src/OrchardCore.Modules/OrchardCore.Users/TimeZone/Services/IUserTimeZoneService.cs

```diff
@@ -11,11 +11,27 @@ public interface IUserTimeZoneService
     /// Gets the time zone for the given user.
     /// </summary>
     /// <param name="user">The <see cref="IUser"/>.</param>
+    [Obsolete("This method is obsolete and will be removed in future release. Instead use GetAsync(userName).")]
     ValueTask<ITimeZone> GetAsync(IUser user);
 
     /// <summary>
     /// Updates the time zone for the given user.
     /// </summary>
     /// <param name="user">The <see cref="IUser"/>.</param>
+    [Obsolete("This method is obsolete and will be removed in future release. Instead use ForgetCacheAsync(userName).")]
     ValueTask UpdateAsync(IUser user);
+
+    /// <summary>
+    /// Gets the time zone for the given user.
+    /// </summary>
+    /// <param name="userName">The user name.</param>
+    ValueTask<ITimeZone> GetAsync(string userName)
```

Comment:
This is not what we said we would, you even had some points to not do that. But move the caching at the timezone selector level.

Reply:
after looking more at the implemenation, I the UserTimeZoneService service becomes useless if we don't handle the cache in there. The UpdateAsync method suggest that it updates the timezone, but it does not do that ever, All it does is invalidate the cache. So it's best to pass the username to the GetAsync method and let it be a wrapper which handles the caching internally incause someone want to use it outside the timeZoneSelector.